packages/dropdown/src/js/__specs__/index.spec.tsx

```diff
@@ -1,62 +1,95 @@
+    expect(formatItemId('aMenuId', undefined, 'aLabel')).toEqual(
+      'aMenuId-aLabel'
     )
-    expect(item).toBeUndefined()
-  })
-
-  it('returns item for matched first-level option', () => {
-    const toFind = <ActionMenu.Item value="findme">Find Me</ActionMenu.Item>
-    const item = findMatchingActionMenuItem(
-      <ActionMenu>
-        <ActionMenu.Item value="diff">Different</ActionMenu.Item>
-        {toFind}
-        <ActionMenu.Item value="morediff">Different</ActionMenu.Item>
-      </ActionMenu>,
-      'findme'
+    expect(formatItemId('aMenuId', undefined, 'a Fine Label')).toEqual(
+      'aMenuId-aFineLabel'
     )
-    expect(item.props.children).toEqual(toFind.props.children)
-    expect(item.props.value).toEqual(toFind.props.value)
-  })
-
-  it('returns item for matched 2nd-level option', () => {
-    const toFind = <ActionMenu.Item value="findme">Find Me</ActionMenu.Item>
-    const item = findMatchingActionMenuItem(
-      <ActionMenu>
-        <ActionMenu.Item value="diff">Different</ActionMenu.Item>
-        <ActionMenu.Item
-          nested={
-            <>
-              <ActionMenu.Item value="stilldiff">Still Diff</ActionMenu.Item>
-              {toFind}
-            </>
-          }
-        >
-          Nested Different
-        </ActionMenu.Item>
-      </ActionMenu>,
-      'findme'
+  })
+})
+
+describe('#parseMenuChildren', () => {
+  it('is empty array if no menu given', () => {
+    expect(parseMenuChildren('aMenuId')).toEqual([])
+  })
+
+  it('parses a fragment of children', () => {
+    const children = (
+      <>
+        <Item value="o">One</Item>
+        <Item value="w">Two</Item>
+        <Item value="h">Three</Item>
+      </>
     )
-    expect(item.props.children).toEqual(toFind.props.children)
-    expect(item.props.value).toEqual(toFind.props.value)
+    expect(parseMenuChildren('aMenuId', children)).toEqual([
+      { id: 'aMenuId-o', label: 'One', value: 'o' },
+      { id: 'aMenuId-w', label: 'Two', value: 'w' },
+      { id: 'aMenuId-h', label: 'Three', value: 'h' }
+    ])
+  })
+
+  it('parses an array of children', () => {
+    const children = [
+      <Item value="o" key="o">
+        One
+      </Item>,
+      <Item value="w" key="w">
+        Two
+      </Item>,
+      <Item value="h" key="h">
+        Three
+      </Item>
+    ]
+
+    expect(parseMenuChildren('aMenuId', children)).toEqual([
+      { id: 'aMenuId-o', label: 'One', value: 'o' },
+      { id: 'aMenuId-w', label: 'Two', value: 'w' },
+      { id: 'aMenuId-h', label: 'Three', value: 'h' }
+    ])
+  })
+})
+
+describe('#getLongestMenuLabel', () => {
+  it('is empty string if no items and no placeholder', () => {
+    expect(getLongestMenuLabel([])).toEqual('')
+  })
+
+  it('is placeholder if no items', () => {
+    expect(getLongestMenuLabel([], 'aPlace')).toEqual('aPlace')
+  })
+
+  it('is placeholder if still longest', () => {
+    expect(
+      getLongestMenuLabel(
```

Comment:
A thing I've noticed is that we have edge cases where the character count isn't enough to determine longest. Sometimes we have words whose character width makes them the longest. Should we address this at some point?

Reply:
Probably.  This has gotten us pretty far. Haven't heard complaints about this aspect thus far.

What are the edge cases?  We could start by recording those in stories/test.